modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -356,7 +356,7 @@ void layoutSpan(GlyphList[] runs) {
     BaseBounds getSpanBounds() {
         if (spanBoundsInvalid) {
             GlyphList[] runs = getRuns();
-            if (runs.length != 0) {
+            if (runs != null && runs.length != 0) {
```

Comment:
on a surface, this check looks ok.

however, in getRuns(), Text:388 we have
```
            /* List of run is initialized when the TextFlow layout the children */
            getParent().layout();
```
meaning that `textRuns` did not initialize even though the `getParent().layout();`  was invoked - why?

Reply:
I mean, are we missing another step?

Reply:
Yes, we are definitely missing something.
I looked at it for quite some time, without a better solution as this check.
If you'd like to investigate further - I would recommend you look into the stack trace of the ticket.
It's related to updating the bounds of the Node.

Reply:
generally speaking - the behavior relating to the bounds is broken.

Some components behave differently, after setting a listener to the bounds.
 1.) Groups tend to call layout in the children.
 2.) And Text seems to call layout on it's parent TextFlow
Both cases are serious problems in my opinion - and this should be designed somehow differently.

But for now, i would be happy if the affected applications don't crash by this bug here.

Reply:
As long as we have some understanding of how `runs` can become null at this point (which I think we do), then the current proposed fix seems fine to me.

Let's also file a follow-up bug to investigate whether we can make the computation of bounds during layout more robust so that we don't have transiently wrong results.

Reply:
I've had a similar problem - solved by explicitly recomputing the layout if it was "dirty".
In this case, it might be ok if the call to getSpanBounds with null runs is followed by another where runs are not null so the client code gets the valid bounds eventually.

But if it's not, the caller ends up with incorrect bounds.